pkg/api/steve/proxy/proxy.go

```diff
@@ -145,12 +147,15 @@ func (h *Handler) authLocalCluster(router *gmux.Router) func(rw http.ResponseWri
 
 func (h *Handler) matchManagementCRDs() gmux.MatcherFunc {
 	return func(req *http.Request, match *gmux.RouteMatch) bool {
-		splitPath := strings.Split(req.URL.Path, "/")
-		if len(splitPath) < 3 {
+		gvr, err := gvrFromPath(req.URL.Path)
+		if err != nil {
+			if errors.Is(err, errBadResource) {
```

Comment:
Two comments here:
- This definitely could use a comment explaining why a bad resource error matches, but other error types don't
- You still need to to check the group name even if we get a bad resource name. For example, there's no reason to match `apis/provisioning.cattle.io/v1/some/thing/that/doesnt/match/at/all`, but we do need to mach the equivalent mgmt resource.

Reply:
I don't agree. You are returning here before you check the mgmt group name (if the err is bad resource you immediately return true, and I don't see you looking for mgmt group in the `gvrFromPath` function).